vllm/model_executor/layers/quantization/quark/quark.py

```diff
@@ -0,0 +1,386 @@
+            return global_quant_config
+
+    def _get_scheme_from_config(self, config: Dict[str, Any]) -> "QuarkScheme":
+        if config.get("output_tensors") or config.get("bias"):
+            raise NotImplementedError(
+                "Currently, Quark models with output_tensors "
+                "and bias quantized are not supported")
+        weight_config = cast(Dict[str, Any], config.get("weight"))
+        input_config = cast(Dict[str, Any], config.get("input_tensors"))
+
+        if self._is_fp8_w8a8(weight_config, input_config):
+            is_fp8_w8a8_supported = self._check_scheme_supported(
+                QuarkW8A8Fp8.get_min_capability(), error=False)
+            if is_fp8_w8a8_supported:
+                weight_qscheme = cast(str, weight_config.get("qscheme"))
+                input_static = (input_config is not None and
+                                not cast(bool, input_config.get("is_dynamic")))
+                return QuarkW8A8Fp8(qscheme=weight_qscheme,
+                                    is_static_input_scheme=input_static)
+        elif self._is_static_tensor_w8a8(weight_config, input_config):
+            weight_qscheme = cast(str, weight_config.get("qscheme"))
+            return QuarkW8A8Int8(qscheme=weight_qscheme,
+                                 is_static_input_scheme=True,
+                                 input_symmetric=input_config.get("symmetric"))
+
+        raise NotImplementedError("No quark compatible scheme was found.")
+
+    def get_scheme(self, layer: torch.nn.Module,
+                   layer_name: str) -> "QuarkScheme":
+
+        layer_quant_config = self._find_matched_config(layer_name, layer)
+
+        # Find the quant_scheme
+        scheme = self._get_scheme_from_config(layer_quant_config)
+        # Raise error if device does not support the scheme
+        # (e.g. fp8 needs ada lovelace)
+        self._check_scheme_supported(scheme.get_min_capability())
+
+        return scheme
+
+    def get_cache_scale(self, name: str) -> Optional[List[str]]:
+        """
+        Check whether the param name matches the format for k/v cache scales
+        in compressed-tensors. If this is the case, return its equivalent
+        param name expected by vLLM
+
+        :param name: param name
+        :return: matching param name for KV cache scale in vLLM
+        """
+        if self.kv_cache_group is None or len(self.kv_cache_group) == 0:
+            return None
+
+        kv_proj_names = [
+            re.split(r"[*.]", kv_cache)[-1] for kv_cache in self.kv_cache_group
+        ]
+        if name.endswith(".output_scale"):
+            if len(kv_proj_names) == 1 and kv_proj_names[0] in name:
+                kv_output_scale_name = "." + kv_proj_names[0] + ".output_scale"
+                return [
+                    name.replace(kv_output_scale_name, ".attn.k_scale"),
+                    name.replace(kv_output_scale_name, ".attn.v_scale")
+                ]
```

Comment:
What is the purpose of returning multiple output names for the same input name? I think if you just return a string like before, you won't need to change all of the call-sites aside from
```
            if (self.quant_config is not None and
                (scale_name := self.quant_config.get_cache_scale(name))):
```

Reply:
Are you suggesting that the k_proj and v_proj here are represented by only one layer, and we can remap the corresponding kv_scale only to the attn.k_scale in the vllm, and then the v_scale can also be assigned through this line of code https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/quantization/kv_cache.py#L56 ? If that's the case, indeed, a one-to-one remapping is sufficient, but we must ensure that this code segment (https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/quantization/kv_cache.py#L56) is not altered.

Reply:
Yes, you found the right place. This is an assumption I expect we will keep and would prefer we keep the duplication internal to the layer initialization, rather than augmenting the weight loading as you've done here. I would appreciate if you made the change

Reply:
ok, thanks. I have made the change.